lib/crash_reporter.js

```diff
@@ -0,0 +1,136 @@
+var logs = require('./logs');
+var os = require('os');
+var packageJson = require('../package.json');
+var preferences = require('./preferences');
+var request = require('request');
+
+// the value of the crash reporter preference
+// the value has to be one of 'on' or 'off'
+var CRASH_REPORTER_PREFERENCE = 'crash.reporter.preference';
+var SUBMIT_CRASH_URL = 'http://tessel-error-reporter.appspot.com/crashes/submit';
+
+// override for testing
+if (process.env.DEV_MODE === 'true') {
+  SUBMIT_CRASH_URL = 'http://localhost:9080/crashes/submit';
+}
+
+// the only instance of the crash reporter
+var crashReporter = null;
+
+var CrashReporter = function() {
+  this.preferences = new preferences.Preferences();
+};
```

Comment:
CrashReporter doesn't need to be a class, since only one instance is ever created and used. Just make a plan object "bag of methods" and export that.